### PR TITLE
[ML-250] Replace sycl::gpu_selector{}.select_device() with manual GPU allocation

### PR DESCRIPTION
**Pull request by minmingzhu**
## What changes were proposed in this pull request?
Used a fixed GPU instead of [select_device ](sycl::gpu_selector{}.select_device()) function

### CODE REVIEW

File: mllib-dal/src/main/native/build-jni.sh [hunk 1]
```diff
@@ -31,13 +31,13 @@ fi
 javah -d $WORK_DIR/javah -classpath "$WORK_DIR/../../../target/classes:$DAAL_JAR:$SPARK_HOME/jars/*" -force \
     com.intel.oap.mllib.OneCCL$ \
     com.intel.oap.mllib.OneDAL$ \
-    com.intel.oap.mllib.clustering.KMeansImpl \
-    com.intel.oap.mllib.feature.PCAImpl \
+    com.intel.oap.mllib.clustering.KMeansDALImpl \
+    com.intel.oap.mllib.feature.PCADALImpl \
```
Comment (xwu99) [inline mllib-dal/src/main/native/build-jni.sh:35]: I think the name should be KMeansImpl now. please check all.

File: mllib-dal/src/main/scala/org/apache/spark/mllib/stat/spark321/Statistics.scala [hunk 2]
```diff
@@ -56,18 +56,19 @@ class Statistics extends SummarizerShim {
...
+        rdd.persist(StorageLevel.MEMORY_AND_DISK)
+        rdd.count()
+      }
       val executor_num = Utils.sparkExecutorNum(X.sparkContext)
       val executor_cores = Utils.sparkExecutorCores()
       val summary = new SummarizerDALImpl(executor_num, executor_cores)
         .computeSummarizerMatrix(rdd)
       if (handlePersistence) {
-        X.unpersist()
+        rdd.unpersist()
```
Comment (xwu99) [inline mllib-dal/src/main/scala/org/apache/spark/mllib/stat/spark321/Statistics.scala:71]: those code changes are irrelevant to this PR?

**Review by xwu99 — CHANGES_REQUESTED** [review]

File: mllib-dal/src/main/native/KMeansImpl.cpp [hunk 3]
```diff
@@ -266,18 +264,14 @@ static jlong doKMeansOneAPICompute(JNIEnv *env, jint rankId, jlong pNumTabData,
...
     auto t1 = std::chrono::high_resolution_clock::now();
     kmeans::train_result result_train =
         preview::train(comm, kmeans_desc, local_input);
     auto t2 = std::chrono::high_resolution_clock::now();
     auto duration =
         std::chrono::duration_cast<std::chrono::milliseconds>(t2 - t1).count();
-    std::cout << "KMeans (native): rankid  " << rankId
-              << "; training step took " << duration / 1000 << " secs"
-              << std::endl;
+    std::cout << "KMeans (native): training step took " << duration / 1000
```
[the remark this answers is not in the document]
Comment (minmingzhu) [reply mllib-dal/src/main/native/KMeansImpl.cpp:273]: Want to output each task training took and compare with root task training.

File: mllib-dal/src/main/scala/org/apache/spark/mllib/stat/spark321/Statistics.scala (hunk 2 above)
Comment (minmingzhu) [reply mllib-dal/src/main/scala/org/apache/spark/mllib/stat/spark321/Statistics.scala:71]: rollback and fix this in next PR

File: mllib-dal/src/main/scala/com/intel/oap/mllib/OneDAL.scala [hunk 4]
```diff
@@ -675,7 +675,6 @@ object OneDAL {
       .coalesce(executorNum,
         partitionCoalescer = Some(new ExecutorInProcessCoalescePartitioner()))
       .setName("coalescedRdd")
-      .cache()
```
[the remark this answers is not in the document]
Comment (minmingzhu) [reply mllib-dal/src/main/scala/com/intel/oap/mllib/OneDAL.scala:678]: rollback and fix this in next PR

**Review by xwu99 — APPROVED** [review]

**Pull request merged**